File: shopbiz-api/src/main/java/self/edu/shopbiz/model/OrderItem.java
package self.edu.shopbiz.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

import static javax.persistence.GenerationType.IDENTITY;

/**
 * Created by mpjoshi on 10/10/19.
 */

@NoArgsConstructor
@Getter
@Setter
@Entity
public class OrderItem {

    @Id
    @GeneratedValue(strategy = IDENTITY)
    private Long id;

    @ManyToOne
    private Product product;

    private Integer orderedQuantities;

    @JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY)
    private Order order;

    private BigDecimal extPrice;

    @CreationTimestamp
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;

}

File: shopbiz-ui/src/app/models/OrderItem.ts
import { CartItem } from './cartItem';
import { Product } from './product';


export class OrderItem {
  id: number;
  product: Product;
  orderedQuantities: number;

  constructor(product: Product, orderedQuantities: number) {
    this.product = product;
    this.orderedQuantities = orderedQuantities;
  }
}
